### PR TITLE
chore: bump Rust version to 1.74.0

### PR DESCRIPTION
**Pull request by xiaguan**
When I wanted to use `clap` version 4 , I found that it requires Rust version >= `1.70.0`. To prevent similar situations in the future, I upgraded to the latest stable version (1.74.0). The upgrade itself will introduce an upgrade to clippy, bringing in new standards.

Clippy warnings mainly occur in the following modules:

1. `fuse`: Issues with byte order calling APIs in test.
2. `etcd_delegate` and `dist`: These need to be thoroughly cleaned up as we no longer require them. We should use `kv_engine` instead.
3. `csi`: This could be maintained as a separate crate or repository.

### CODE REVIEW

**Comment by xiaguan** [comment]
Related issues: #434  #433

File: macro-utils/src/lib.rs [hunk 1]
```diff
@@ -78,7 +78,7 @@
     clippy::shadow_unrelated, //it’s a common pattern in Rust code
     clippy::shadow_reuse, //it’s a common pattern in Rust code
     clippy::shadow_same, //it’s a common pattern in Rust code
-    clippy::integer_arithmetic,  // An error is raised when calling `quote::quote!`
+    clippy::arithmetic_side_effects,  // An error is raised when calling `quote::quote!`
     clippy::arithmetic_side_effects  // As same as above
```
Comment (Reto911) [inline macro-utils/src/lib.rs:82]: Why do you amend the line 81 to `clippy::arithmetic_side_effects` while there is a `clippy::arithmetic_side_effects` below already?

File: src/common/error.rs [hunk 2]
```diff
@@ -290,16 +290,16 @@ where
     where
         C: Into<String>,
     {
-        self.map_err(|e| e.into().add_context(ctx))
+        self.map_err(|err| err.into().add_context(ctx))
```
Comment (Reto911) [inline src/common/error.rs:293]: Which clippy lint did this violate? Does it check an identifier with single letter?

File: src/async_fuse/fuse/fuse_reply.rs [hunk 3]
```diff
@@ -24,7 +24,7 @@ use super::protocol::{
 #[cfg(feature = "abi-7-18")]
 use super::protocol::{FuseNotifyCode::FUSE_NOTIFY_DELETE, FuseNotifyDeleteOut};
 
-/// This trait describes a type that can be converted to Vec<IoVec<&[u8]>>
+/// This trait describes a type that can be converted to Vec<`IoVec`<&[u8]>>
```
Comment (Reto911) [inline src/async_fuse/fuse/fuse_reply.rs:27]: Which clippy lint did this violate? Shall we say `Vec<IoVec<&[u8]>>` here?

File: src/async_fuse/fuse/fuse_reply.rs [hunk 4]
```diff
@@ -35,7 +35,7 @@ pub trait AsIoVecList {
 }
 
 /// Any type implement the `AsIoVec` trait, its Vec can be automatically for
-/// Vec<IoVec<&[u8]>>.
+/// Vec<`IoVec`<&[u8]>>.
```
Comment (Reto911) [inline src/async_fuse/fuse/fuse_reply.rs:38]: Same as above

File: src/async_fuse/memfs/serial.rs [hunk 5]
```diff
@@ -79,7 +79,7 @@ pub enum SerialSFlag {
 }
 
 /// In order to derive Serialize and Deserialize,
-/// Replace the 'BTreeMap<String, `DirEntry`>' with 'HashMap<String,
+/// Replace the `BTreeMap`<String, `DirEntry`>' with `HashMap`<String,
```
Comment (Reto911) [inline src/async_fuse/memfs/serial.rs:82]: Same as above

File: src/async_fuse/fuse/fuse_reply.rs (hunk 3 above)
Comment (xiaguan) [reply src/async_fuse/fuse/fuse_reply.rs:27]: Clippy did suggest me to do it.

Comment (Reto911) [reply src/async_fuse/fuse/fuse_reply.rs:27]: Sorry, I mean, add the backticks (\`) outside the whole `Vec<IoVec<&[u8]>>`, as it is a complete type.

Comment (xiaguan) [reply src/async_fuse/fuse/fuse_reply.rs:27]: I think current way is better. Cause `IoVec` is defined in this mod.

**Review by Reto911 — APPROVED** [review]

**Pull request merged**